pki/services.go

```diff
@@ -492,10 +504,17 @@ func GenerateKubeletCertificate(ctx context.Context, certs map[string]Certificat
 		return fmt.Errorf("CA Certificate or Key is empty")
 	}
 	log.Infof(ctx, "[certificates] Generating Kubernetes Kubelet certificates")
+	unusedKubeletCerts := make(map[string]bool)
+	for k := range certs {
```

Comment:
These changes look identical to the etcd ones. Can we extract a common function to be invoked in these two places

Reply:
> The issue/pr description is talking about Rancher APIs. Can you justify why the fix needs to be on the RKE side?
  justified in:https://github.com/rancher/rancher/issues/24333